### ISSUE TITLE
flexible notification option

### ISSUE THREAD

**Issue by adipo** [vitalidze/traccar-web#153]
Hello,

It is possible to extend notification menu with something more flexible like an http request or launch an application?

For notification we have email and Pushbullet as options. For me it would be easy to start applications or to send http commands.

This way I could follow watchman making his guard tour on a CCTV system or I could detect presence of a peron in a dangerous area and switch off some dangerous equipaments.

With http commans and some dedicated equipaments it is posiible to send sms or make all kind of automations.

Thank you

**Comment by vitalidze** [comment]
I think it is possible to add third notification method - call the HTTP url. Will add it once I finish tasks with higher priority.

**Comment by vitalidze** [comment]
We have to think how it will correlate with fact that notification is now sent per user. So same event may be sent to several users. I guess that same should be applied for HTTP request, in which case some user information (email, username, cell phone?) may be included.

**Comment by adipo** [comment]
Hello Vitaly,

I'm glad you consider my proposal.

The http request should be linked with device and geofence or device behaviour (speed, connection etc) not with the user.

I suggest to add some fields in Geo-Fence/Devices window (see attached image).

![image](https://cloud.githubusercontent.com/assets/12082712/7319351/95cb2470-ea9d-11e4-86fa-463c7f5081bc.png)

Best regards,

**Comment by Abyss777** [comment]
I think it is not very good idea to write such many URLs (if I have 10000 devices?). The better way is to write one URL in Notification configuration and send parameters. Something like this:

https://notify.traccar/?deviceid=00000001&status=entergeo&geofencename=parking&... 

Web service should process parameters and do its job.

**Comment by vitalidze** [comment]
I agree with @Abyss777 , I thought about implementing same placeholders in URL as for usual email/pushbullet notifications. Then you can just create an URL per device and routing may be performed on some third party web service if necessary.

**Comment by adipo** [comment]
I was thinking about connecting  with hardware (with http request features). In this case the device is not able to use parameters from traccar. The http mesage should parameters for the device behaviour.

This might be useful in security and life safety projects where there are not many targets (devices) but it is very useful to have a link between position and automation systems.

**Comment by Unr3al2** [comment]
Super,
That's what I meant.
I watched topic :)

**Comment by Abyss777** [comment]
Hello
Is there any progress with notifications via HTTP requests?
I want to do some integration to Smart-Home system. The easiest way is to do it with GET requests.
It will help to automate some tasks if Smart-Home will know about entering/leaving geofences.

**Comment by vitalidze** [comment]
@Abyss777 There are no progress at the moment.
